crates/oxc_transformer/src/commonjs/utils/define.rs

```diff
@@ -0,0 +1,145 @@
+//! It is not `amd define`, but for `Object.defineProperty` related codes.
+
+use oxc_ast::ast::{
+    BindingRestElement, Expression, FormalParameterKind, FunctionType, IdentifierReference,
+    PropertyKind, Statement, TSThisParameter, TSTypeAnnotation, TSTypeParameterDeclaration,
+    TSTypeParameterInstantiation,
+};
+use oxc_ast::AstBuilder;
+use oxc_span::SPAN;
+use oxc_syntax::identifier;
+use std::borrow::Cow;
+
+/// Generates `Object.defineProperty` call for a given target.
+///
+/// Useful when re-exporting:
+///
+/// ```js
+/// export { foo as bar } from './foobar.js'
+/// ```
+///
+/// It can be transformed from:
```

Comment:
CJS to ESM?  Is this a typo?

Reply:
I don't think it is a typo (?) transform `export ... from ...` to `Object.defineProperty(...)`, it is what the function does.